doc/md/jdk16-access-exceptions.md

```diff
@@ -0,0 +1,62 @@
+# JDK 16+ access exceptions
+
+### Problem symptoms 
+On JDK 16 and above you may encounter `InaccessibleObjectException` or `IllegalAccessException` in following known cases:
+1. _Some_ usages of `mockkStatic` on Java standard library classes, e.g. 
+   ```kotlin
+   mockkStatic(Instant::class)
+   every { Instant.now() } returns Instant.parse("2022-09-15T09:31:26.919Z")
+   ```
+   ```
+   java.time.format.DateTimeParseException: Text '2022-09-15T09:31:26.919Z' could not be parsed:
+     Unable to make private static java.time.Instant java.time.Instant.create(long,int) accessible: 
+     module java.base does not "opens java.time" to unnamed module @4501b7af
+   ```
+
+2. Spying on Java standard library classes, e.g.
+   ```kotlin
+   val socket = Socket()
+   val spy = spyk<Socket>(socket)
+   ```
+    Note: this includes creating `@SpyKBean` on spring data repository instances, because they are instances of `java.lang.reflect.Proxy` in fact.
+   ```
+   Error creating bean with name 'featureRepository': Post-processing of FactoryBean's singleton object failed;
+    nested exception is java.lang.IllegalAccessException: class io.mockk.impl.InternalPlatform 
+    cannot access a member of class java.lang.reflect.Proxy (in module java.base) with modifiers "protected"
+   ```
+
+### Problem cause
+JDK 16 enforced strong encapsulation of standard modules. In practice this means you (and your libraries) can no more get away with call like `privateMember.setAccessible(true)` to any of JDK classes.
+
+https://blogs.oracle.com/javamagazine/post/a-peek-into-java-17-continuing-the-drive-to-encapsulate-the-java-runtime-internals
+
+### Solution / workaround
+Add JVM argument `--add-opens java.base/java.time=ALL-UNNAMED` _for each package_ you need  
+Description: https://docs.oracle.com/en/java/javase/16/migrate/migrating-jdk-8-later-jdk-releases.html
+
+For Gradle users:
+```groovy
+tasks.test {
+    jvmArgs("--add-opens", "java.base/java.time=ALL-UNNAMED")
+}
+```
```

Comment:
I am not very familiar with gradle, and copied this from one of the issues. May need help from experienced gradle user to verify that this snippet looks ok

Reply:
Yeah it looks okay. It will only change the 'test' class though. In .kts, this is the best way to change all test tasks:

```kts
// build.gradle.kts

tasks.withType<Test>().configureEach {
  jvmArgs(...)
}
```